pkg/controller/installation/core_controller.go

```diff
@@ -1017,11 +1017,9 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 		}
 
 		managementCluster, err = utils.GetManagementCluster(ctx, r.client)
-		if managementCluster != nil {
```

Comment:
I think this was just a bug - `managementCluster` would never be non-nil if `err` was also non-nil, so we were just swallowing errors from this function.